filebeat/input/filestream/internal/input-logfile/manager.go

```diff
@@ -229,7 +231,19 @@ func (cim *InputManager) Create(config *conf.C) (v2.Input, error) {
 	}, nil
 }
 
-// StopInput peforms all necessary clean up when an input finishes.
+func (cim *InputManager) Delete(cfg *conf.C) error {
+	settings := struct {
+		ID string `config:"id"`
+	}{}
+	if err := cfg.Unpack(&settings); err != nil {
+		return fmt.Errorf("could not umpack conig to get inut ID: %w", err)
+	}
+
+	cim.StopInput(settings.ID)
```

Comment:
We need to check for an empty string before this.

Reply:
I don't see a point in adding the empty ID to the map.
Overriding will lead to an input running indefinitely without being closed. So, maybe we need another list to account for all the inputs then? I think this map is only valid for accounting unique IDs not the running inputs.

Reply:
>I don't see a point in adding the empty ID to the map.

Neither do I but it's the current behaviour, so the `Delete` is just complying with it.

>I think this map is only valid for accounting unique IDs not the running inputs.

Yes, that seems to be the case.

>Overriding will lead to an input running indefinitely without being closed. So, maybe we need another list to account for all the inputs then? 

Agreed, I found it a bit odd how it works, but I did not investigate how the `ids` map is used on other parts of the code. But yes, I believe it worth investigating is there might be a bug there. However, I'd say it's for another Issue.

Reply:
> However, I'd say it's for another Issue.

Could you please create an issue and link it here?

Reply:
Thus an empty ID id as valid as any other and suffer from the same uniqueness problem.
@rdner do you agree?

Reply:
@AndersonQ I don't really understand your comment but I trust @belimawr on this one, he dealt with that in the past.